types/index.d.ts

```diff
@@ -0,0 +1,565 @@
+   * @default undefined
+   */
+  hda?: V86Image
+
+  /** 
+   * First floppy disk, see below.
+   * @default undefined
+   */
+  fda?: V86Image
+
+  /** 
+   * cdrom
+   * @default undefined
+   */
+  cdrom?: V86Image
+
+  /**
+   * A Linux kernel image to boot (only bzimage format)
+   * @default undefined
+   */
+  bzimage?: V86Image
+
+  /**
+   * A Linux ramdisk image
+   * @default undefined
+   */
+  initrd?: V86Image
+
+  /**
+   * Automatically fetch bzimage and initrd from the specified `filesystem`.
+   */
+  bzimage_initrd_from_filesystem?: boolean
+
+  /**
+   * An initial state to load
+   * @default undefined
+   */
+  initial_state?: V86Image
+
+  /**
+   * A 9p filesystem
+   * @default undefined
+   */
+  filesystem?: Filesystem9pOptions
+
+  /** 
+   * A textarea that will receive and send data to the emulated serial terminal.
+   * Alternatively the serial terminal can also be accessed programatically, see [serial.html](../examples/serial.html).  
+   * @default undefined
+   */
+  serial_container?: HTMLTextAreaElement
+
+  /**
+   * Xtermjs serial terminal container. When set, serial_container option is ignored.
+   * @default undefined
+   */
+  serial_container_xtermjs?: HTMLElement
+
+  /**
+   * An HTMLElement. This should have a certain structure, see [basic.html](../examples/basic.html).
+   * @default undefined
+   */
+  screen_container?: HTMLElement | null
+
+  /**
+   * ACPI
+   * @default false
+   */
+  acpi?: boolean
+
+  /**
+   * log level
+   * @default LogLevel.LOG_NONE
+   */
+  log_level?: LogLevel
+
+  /**
+   * boot order
+   * @default BootOrder.CD_FLOPPY_HARDDISK
+   */
+  boot_order?: BootOrder
+
+  /**
+   * fast boot
+   * @default false
+   */
+  fastboot?: boolean
+
+  /**
+   * enables UART1 (Serial port)
+   * @see http://wiki.osdev.org/UART
+   * @see https://github.com/s-macke/jor1k/blob/master/js/worker/dev/uart.js
+   * @see https://www.freebsd.org/doc/en/articles/serial-uart/
+   * @default undefined
+   */
+  uart1?: boolean
+
+  /**
+   * enables UART2 (Serial port)
+   * @see http://wiki.osdev.org/UART
+   * @see https://github.com/s-macke/jor1k/blob/master/js/worker/dev/uart.js
+   * @see https://www.freebsd.org/doc/en/articles/serial-uart/
+   * @default undefined
+   */
+  uart2?: boolean
+
+  /**
+   * enables UART3 (Serial port)
+   * @see http://wiki.osdev.org/UART
+   * @see https://github.com/s-macke/jor1k/blob/master/js/worker/dev/uart.js
+   * @see https://www.freebsd.org/doc/en/articles/serial-uart/
+   * @default undefined
+   */
+  uart3?: boolean
+
+  /**
+   * boot cmdline 
+   */
+  cmdline?: string
+
+  /**
+   * Ne2k: should MAC be preserved from the state image
+   * @default undefined
+   */
+  preserve_mac_from_state_image?: boolean  
+
+  /**
+   * custom network adapter
+   * @default undefined
+   */
+  network_adapter?: NetworkAdapter
+
+  /**
+   * enables screen dummy
+   * @default undefined
+   */
+  screen_dummy?: boolean
+}
+
+declare class V86Starter {
```

Comment:
This should be called V86, V86Starter is on its way out (would also be good to rename V86StarterOptions and other mentions).

Reply:
@copy: I am able to rename the interfaces, not the class declarations, or do you wish to rename the `V86Starter` to `V86` inside the codebase?